### ISSUE TITLE
Update limited medical authorizations in the 21-22 and 21-22a form generation

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#98295]
## Background

This pertains to [Field 20 on form 21-22](https://www.vba.va.gov/pubs/forms/VBA-21-22-ARE.pdf) and [Field 21 on form 21-22a](https://www.vba.va.gov/pubs/forms/vba-21-22a-are.pdf). While the frontend of Appoint a Rep will send over what medical records the user authorizes access to, **the forms need handle this input with an opposite approach**:
- Form 21-22 should check the boxes of whatever IS NOT authorized
- Form 21-22a should notate in the free-text area whatever IS authorized.

We've talked about some updates to our PDF generation, to make this extra clear.

#### 21-22
We need take the opposite approach and check the boxes of whatever IS NOT authorized (currently we are checking the boxes for what is authorized). Scenario example:
> A Veteran authorizes access to Drug Abuse records, but declines access to records for Alcoholism, HIV, and Sickle Cell Anemia.
> 
> 1. Field 19 is checked  to indicate general authorization.
> 2. Field 20 has the boxes for Alcoholism, HIV and Sickle Cell Anemia checked to indicate exceptions to the authorization. The Drug Abuse box remains unchecked to indicate those records are authorized.

#### 21-22a
Field 21 on form 21-22a is a bit ambiguous on what "limited" pertains too -- The correct answer is that the rep is authorized to access whatever conditions are listed in this field, so how we fill out this field could use a small update to make this clear.

Currently, we are simply listing the records that the user is authorizing access to. 
> Alcoholism and alcohol abuse records, Drug abuse records, and Sickle cell anemia records.

To make the intent extra clear, we would like to add a phrase before the medical records list:
> **I authorize access to:** Alcoholism and alcohol abuse records, Drug abuse records, and Sickle cell anemia records.

_Note: It'd also be nice to end with a period, for proper grammar._

## Tasks
21-22:
- [x] Take the opposite approach and check the boxes that the user DOES NOT authorize.

21-22a:
- [x] Add a phrase before the medical conditions 
- [x] Add a period at the end of the medical conditions

## Acceptance Criteria
- [x] Field 20 of the generated 21-22 PDF accurately reflects which medical authorizations the user has NOT provided access to.
- [x] Field 21 of the generated 21-22a PDF is extra clear for which medical authorizations the user has provided access to.

**Comment by opticbob** [comment]
Some updates on this:

- I got a tip on a helper that can be used to see if all field values match: spec/support/pdf_fill_helper.rb.
- That didn't work.
- A little investigation led me to realize that our finished product PDFs have no fields defined at all.  This is probably related to the accessibility finding that our PDFs aren't tagged at all.
- I disabled flattening in the PDF form fill, which allows further editing of the form.
- This restores the keys but makes the form display weirdly in Acrobat.  It's fine in chrome.
- There was some font display weirdness in Acrobat last week as well, but no changes in the underlying font files we're using for the Next Steps page on the PDF.
- I confirmed that the last time I made PDF changes 3 weeks ago everything displayed fine in Acrobat.
- I'm working now on finding out where our Acrobat display issues started to see if I can get the PDF with keys defined to display in Acrobat correctly.

**Comment by opticbob** [comment]
With help from @coforma-jamie we determined that the unflattened and editable PDFs are still not accessible so for this ticket I'm going to implement a hybrid testing strategy.  We'll continue comparing the text of PDFs like we do now but we'll also test unflattened PDFs so we can compare their form fields directly.  This will allow us to test the status of checkboxes and let us know if we break something with checkboxes in the future.  I've reorganized the existing fixture PDFs to make adding unflattened fixture PDFs easier.  I'll start generating those unflattened fixture PDFs tomorrow.
